### ISSUE TITLE
Merge use-contextlib-suppress and try-except-pass

### ISSUE THREAD

**Issue by not-my-profile** [charliermarsh/ruff#2714]
As per the rule naming convention established in #2712 we do not want any rule names starting with use-*. SIM105 is currently named "use-contextlib-suppress" violating the naming convention. I think the proper rule name would be "try-except-pass" but we cannot simply rename SIM105 to that since that name is already used by S110.

Currently neither of these rules support autofix, so we can easily merge them and just mention both solutions in the documentation. In the long run it would be nice if ruff supported different autofixes for one violation, I just opened #2713 for tracking that.

**Comment by charliermarsh** [comment]
Agreed. Not sure which to "remove" (i.e., treat as the canonical rule, as opposed to the redirect), but I guess that gets back to the discussion in #2517.

**Comment by not-my-profile** [comment]
I don't think it matters much which code we map and which one we redirect for now, since #2517 should land soon(tm).

**Comment by charliermarsh** [comment]
To unify blind- and bare-except, we need to change `bare-except` to flag `except Exception` and `except BaseException`.

**Comment by not-my-profile** [comment]
Ah right ... good catch! I think `except:` and `except BaseException:` are equivalent since every exception must derive `BaseException`. `except Exception` is different though since some exceptions e.g. `GeneratorExit`, `KeyboardInterrupt` and `SystemExit` don't derive from it. So we may want to split `blind-except` into `except-any` and `except-exception`.

**Comment by charliermarsh** [comment]
We could also make it a setting on the rule, rather than creating separate rules. But I don’t think we’ve articulated a strict philosophy either way there.

**Comment by andersk** [comment]
use-contextlib-suppress (SIM105) is intentionally much weaker than try-except-pass (S110), and it’s not unreasonable to enable the former without the latter. See #1948.
